src/python/WMCore/WorkQueue/WorkQueueBackend.py

```diff
@@ -524,3 +524,41 @@ def getWMBSInjectStatus(self, request = None):
                     finalInjectionStatus.append({element._id : False})
 
             return finalInjectionStatus
+        
+    def getWorkflowNames(self, inboxFlag = False):
+        """Get workflow names from workqueue db"""
+        if inboxFlag:
+            db = self.inbox
+        else:
+            db = self.db
+        data = db.loadView('WorkQueue', 'elementsByWorkflow', 
+                           {'stale': "update_after", 'reduce' : True, 'group' : True})
+        return [x['key'] for x in data.get('rows', [])]
+    
+    def deletWQElementsByWorkflow(self, workflowNames):
+        """
+        delete workqueue elements belongs to given workflow names 
+        it doen't check the status of workflow so need to be careful to use this.
+        Pass only workflows which has the end status
+        """
+        deleted = 0
+        dbs = [self.db, self.inbox]
+        if type(workflowNames) != list:
+            workflowNames = [workflowNames]
+        
+        if len(workflowNames) == 0:
+            return deleted
+        
+        options = {} 
+        options["stale"] = "update_after"
+        options["reduce"] = False
+        
+        for couchdb in dbs:
+            result = couchdb.loadView("WorkQueue", "elementsByWorkflow", options, workflowNames)
```

Comment:
Is the doc "id" the same as doc "key". If so, then we don't need to query this view again, since we get these results in getWorkflowNames()